### ISSUE TITLE
Add beacons to skystone chests to allow for compass enhancements.

### ISSUE THREAD

**Issue by isochronous** [AppliedEnergistics/Applied-Energistics-2#2055]
This is a bit similar to the rejected #2051 and #550 suggestions, but unlike those feature requests, makes a lot more sense from a scientific perspective.

Whatever civilization sent out these meteors, they almost certainly would want a way to locate them again, right? So it would make sense for each skystone chest to contain some type of beacon. That in itself wouldn't help the player much, because they obviously wouldn't have a receiver or locator for those beacons, but once we got our blunt little hands on one, I bet we could figure out a way to modify the skystone compass to point to it rather than the skystone itself.

I was trying to come up with some ideas on how the compass could be modified, or a new one built from scratch that's tuned to the beacons.  I was thinking that you could, for example, require four charged certus crystals surrounding a beacon, which would make a beacon receiver (or whatever you want to call it). You'd then craft your skystone compass with the beacon receiver and get a skystone beacon locator, which would work just like the compass only be tuned to the beacons in the chests. 

The idea here, basically, is to make the process of locating different meteors just a little bit easier, but only after you've found the first one with just the compass. It would also make perfect sense from a scientific perspective, which seemed to be the issue with my previous suggestion.

Other possibilities:
- have the graphic for the beacon locator include a little resistor-looking thing fused to the top of the compass itself, and put a little light on it that flashes faster the closer you get to a beacon. 
- put four indicators (lights, alien symbols, runes, whatever) on the beacon that correspond to the four types of press plate, and have them light up, pulse, wiggle, or raise/lower depending on what type(s) of plates are in the chest with the beacon. This could be something that requires a further enhancement to the beacon locator, just to make it more expensive in exchange for the extra functionality. Actually, since the locator was originally a transmitter rather than a receiver, that particular functionality would almost certainly require further modifications to work. I think this is the idea least likely to be implemented though, so I'll leave possible recipes up to you in the off chance you decide to implement it.

**Comment by isochronous** [comment]
Actually, it might make more sense to create a beacon-tuned crystal that's similar to a charged certus quartz crystal, in that it's created by putting a regular certus quartz crystal in a block of water with the beacon. You'd then have to make a lead-encased beacon (or faraday-caged beacon) by crafting 8 lead ingots (or 8 copper ingots) surrounding your beacon to prevent the beacon-tuned crystal from always pointing at the beacon in your possession. This would mean the proximity-light & press plate indicator ideas wouldn't make sense, or would have to be accomplished in a different manner. It could perhaps be done by disassembling the beacon by crafting it with a screwdriver, which would yield three or four components that could then be used to further enhance the locator.

**Comment by yueh** [comment]
There is really not gameplay reason to improve the compass in any way. It already provides way more convenience than vanilla minecraft itself as you still need to find a couple of diamonds, gold and other materials and there is no compass for these things.

Even with just walking around and no sprinting, it is easy to encounter a new meteor every few minutes.
With swiftness/running/ender pearls or any other walk speed improvment, it is probably very easy to find one every 30-45 seconds. So it is really easy to find all presses within a single minecraft day and I doubt that 1 minecraft day is enough time to find enough diamonds, gold or iron to last forever like presses.

Should someone still not like it, they can simply disable meteors and/or make the presses craftable with AE2. The default config is just as it says a default one. Nobody is forced to used it and can change it to fit their particular needs.

**Comment by isochronous** [comment]
Funny, because I was playing the game, and I definitely felt a need for a compass improvement. Sure, there's no compass for diamonds or gold ore in vanilla Minecraft, but neither is there an energy to mass storage system in vanilla Minecraft.  In addition, press plates are far more like TiC casting molds than like gold or diamonds, and they aren't consumable items either.

 It was NOT as easy as just walking around for me to locate all of the press pieces, because my spawn was on a large island that only had one meteor on it, and it took me five meteors to locate all of the press pieces.  If that's your argument, why do you have a compass at all? Why provide an item that helps you locate meteors if it's just a matter of "walking around?"

And I'm sorry, but finding a meteor every 30-45 *seconds*? Are you joking? Even flying around in creative mode it took me more than five minutes to locate a second meteor, because I first had to get far enough away from the first one for the compass to stop pointing at it. As I mentioned before, I tried eliminating all of the skystone from two meteor locations - a process that takes about an hour if playing legit - and even when I couldn't find a single block remaining the compass just kept spinning and spinning.

Rejecting a feature request because it doesn't make logical sense within the context of the game is one thing, but telling users that there's no gameplay reason to improve the compass is just arrogant. I specialize in interaction design, UI design, and user experience, and if I were to tell not just one but several users that "there's no need" for an improvement that they'd all suggested, I'd be out of a job. There's already been more than one request for the compass to work differently, and in those previous requests the objection was always that it didn't make logical sense for it to work that way. In addition, more than one AE2 showcase video I've watched has called the press plate finding process a pain in the ass because of the way the compass works, as have people discussing the mod on Reddit. Now I've suggested a way to improve it that IS logical, would still require an investment of time, effort, and resources in the game, and now the counter-argument has changed to "there's no reason to do that." Well, one of the first things I learned in my profession is that when multiple people are asking for the same thing, it's because *they feel a need for it.*

**Comment by yueh** [comment]
Meteors are at least about 700 blocks apart, maybe a bit more due to RNG.
Walkspeed in minecraft is around 4-8 blocks/sec (swiftness, sprinting etc), afaik around 20 with ender pearls. So even when assumimg about 1000-1500 blocks distance between them, it should be easily possible to visit one every few minutes. 
Each time I started a world, it usually took around 1 to maybe 1.5 minecraft days (20-30 minutes) to find all presses. Was mostly depending on how many underground meteors were generated and I do it usually pretty early one even food is usually sparse, thus not much running. Usually I spent way more time mining or find the first charged certus.
Just skip it and walk like 300-400 blocks away from it and the compass should pick up a new one. 

Regarding the gameplay argument (this includes balancing etc in my opinion), we have an idea about how difficult it should be. And ridiculuous easy is not one of it, thus any "improvment" to the compass was usually rejected because if it.
AE2 is still more of an end game mod, maybe somewhat midgame (by default). Other mods lock their endgame parts behind way more harder methods. Factorization for example looks it behind dungeon chests. No compass, way harder to find and I had worlds without finding a single one. Some require emeralds. Which only spawn in a single minecraft biome and extremely rare. Last time I dug 2 or 3 3x3 tunnels of about 1k block length through it to find a single one with a couple of trips back and forth to my base to unload stuff (Like 1.5k blocks way and also an island). I assume you can imagine how much time it did take.

The main point here is really, this is not an improvement in terms of usability or similar.
Which would be a valid suggestion. Say something like "the compass is a bit too small to be easily visible" or "it is not clear, in which direction it points when the meteor is directly in front of your". Or stuff like colorblindness
This is just a request to dumb it down. Which by default is already rejected in our [suggestion guidelines](http://ae-mod.info/Suggestion-Guidelines/).
Games simply have difficulty as an actual feature, which is nothing you will never have as feature in any business application. So will rejecting any usability improvment for them would be stupid, but it is not really a case for a game. Usually it is also always a request to make it easier, but I never saw a request to make it more difficult as the player thinks you need to at least grind for two real life days for any medium game achievement. So if you want to compare it with a business application, you should really think of an user requesting to remove a feature, because they never use it and do not want to see that button for it. While the majority of the other users are using it but would never consider requesting it to stay.

Of course we understand that some players like it easier (and some harder). So as already recommended you can simple add a [custom recipe](http://ae2.ae-mod.info/Custom-Recipes/) and make these craftable.
Which is one of the reason AE2 has a pretty flexibile config and any player can change the recipes to their requirements. The default config is just what we consider as balance for AE2, but this will never satisfy every possible need/play/modpack/etc.
